### PR TITLE
Added new game state to accommodate Turbo swing

### PR DESCRIPTION
**Pull request by blackshadowshade**
Addresses part of #91.

http://jenkins.buttonweavers.com:8080/job/buttonmen-blackshadowshade/700/

### CODE REVIEW

File: src/engine/BMSkillTurbo.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     * @return string
+     */
+    protected static function get_description() {
+        return 'After your starting roll, you may change the size of a ' .
+               'Turbo Swing die or a Turbo Option die every time you roll it. ' .
+               'Decide on a size first that is valid for the Swing or Option '.
+               'type, then roll the new die as usual. If a Turbo Die is ' .
+               'rerolled by a player other than the player who has the Turbo ' .
+               'die (e.g., during a trip attack), then the size cannot be ' .
+               'changed.';
```
Comment (cgolubi1) [inline src/engine/BMSkillTurbo.php]: The phrasing of this description feels a little awkward to me, but i don't have a better suggestion offhand.  If anyone wandering by wants to take a stab at wordsmithing, go for it.  Otherwise i'll try to come up with some concrete suggestions in the near future.

File: src/engine/BMSkillTurbo.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    /**
+     * Descriptions of interactions between this skill and other skills
+     *
+     * An array, indexed by other skill name, whose values are descriptions of
+     * interactions between the relevant skills
+     *
+     * @return array
+     */
+    protected static function get_interaction_descriptions() {
+        return array();
```
Comment (cgolubi1) [inline src/engine/BMSkillTurbo.php]: IMO, the comment you put in the main description about the interaction between Turbo and Trip should be listed as an interaction description between Turbo and Trip.

File: test/src/engine/BMGameTest.php [hunk 3]
```diff
@@ -5759,6 +5773,10 @@ public function test_berserk_round() {
         $this->assertGreaterThanOrEqual(1, $game->actionLog[1]->params['postAttackDice']['attacker'][0]['value']);
 
         $game->update_game_state();
+        $this->assertEquals(BMGameState::CHOOSE_TURBO_SWING, $game->gameState);
```
Comment (cgolubi1) [inline test/src/engine/BMGameTest.php]: Is it expected that games without turbo dice should move through `BMGameState::CHOOSE_TURBO_SWING`?  I didn't think games without chance/focus dice moved through `BMGameState::REACT_TO_INITIATIVE`, but maybe i'm wrong about that.

**Comment by cgolubi1** [comment]
Okay, i made various code comments above, but, modulo those, the upshot of this is really just that i should run some regression tests of arbitrary games and declare victory.

File: test/src/engine/BMGameTest.php (hunk 3 above)
Comment (blackshadowshade) [inline test/src/engine/BMGameTest.php]: At the moment, every game goes through every game state that exists, and if the game state is irrelevant, the code moves on to the next game state.

File: src/engine/BMSkillTurbo.php (hunk 1 above)
Comment (cgolubi1) [inline src/engine/BMSkillTurbo.php]: Okay, so, why does this seem awkward to me?  Basically it just feels too wordy for the amount of information:
* why say "a Turbo Swing die or a Turbo Option die" rather than "a Turbo Swing or Option die"?
* "Decide on a size first that is valid for the Swing or Option type, then roll the new die as usual" - this feels both too verbose and insufficiently informative.  I feel like it sort of implies that you have to take an action to reroll the die after "deciding on a size".  Maybe something more like "Before each reroll, you will get the chance to select any valid size for the Swing or Option type"?
* "If a Turbo Die is rerolled by a player other than the player who has the Turbo die" - again, this seems awkward.  Do we really not have a more concise phrasing for "the player who owns a die"?

**Comment by cgolubi1** [comment]
I don't see any regressions due to the logic changes, so i want to iterate on the wording for the description, add the interaction description between Turbo and Trip, and then this should be good to go.

**Comment by blackshadowshade** [comment]
@jimmosk, how would you rework the wording?

**Comment by blackshadowshade** [comment]
Let's give this a go.

Base description:
After your starting roll, you may change the size of a Turbo Swing or Option die every time you roll it. Decide on a size first that is valid for the Swing or Option type, then roll the new die as usual.

Interactions with Boom, Trip
If a Turbo Die is rerolled because it is the target of a Boom/Trip attack, then the size cannot be changed.

Would these be okay?

**Comment by cgolubi1** [comment]
On Fri, 08 May, 2015 at 15:23:59 -0700, James Ong wrote:

> Let's give this a go.
> 
> Base description:
> After your starting roll, you may change the size of a Turbo Swing or Option die every time you roll it. Decide on a size first that is valid for the Swing or Option type, then roll the new die as usual.
> 
> Interactions with Boom, Trip
> If a Turbo Die is rerolled because it is the target of a Boom/Trip attack, then the size cannot be changed.
> 
> Would these be okay?

Sure, i'd accept these.

Chaos

**Comment by blackshadowshade** [comment]
http://jenkins.buttonweavers.com:8080/job/buttonmen-blackshadowshade/718/ (if it passes)

**Comment by cgolubi1** [comment]
Okay.  Thanks.
